### PR TITLE
Last feature: Add "join table" for visits

### PR DESCRIPTION
**Pull request by pasytchangwa**
In this milestone the following was done:

- Updated schema.sql file
- Updated data.sql file
- Updated queries.sql

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -38,3 +38,11 @@ WHERE Name = 'Charmander' or Name = 'Squirtle' or Name = 'Blossom';
 UPDATE animals
 SET owner_id = 5
 WHERE Name = 'Angemon' or Name = 'Boarmon';
+
+INSERT INTO vets (name,age,date_of_graduation) VALUES('Willian Tatcher', 45, (to_date('04/23/2000','mm/dd/yyyy')) ),('Maisy Smith', 26, (to_date('01/17/2019','mm/dd/yyyy')) ),('Stephanie Mendez', 64, (to_date('05/04/1981','mm/dd/yyyy')) ),('Jack Harkness', 38, (to_date('06/08/2008','mm/dd/yyyy')) );
```
Comment (Emmyn5600) [inline data.sql]: - [x] Let us use this way to insert the data into vets table to keep our code clean and clear as well as easy to read.

  `INSERT INTO vets (name, age, date_of_graduation)
   VALUES ('William Tatcher', 45, '2000-04-23');`

  ` INSERT INTO vets (name, age, date_of_graduation)
   VALUES ('Maisy Smith', 26, '2019-01-17');`

  ` INSERT INTO vets (name, age, date_of_graduation)
   VALUES ('Stephanie Mendez', 64, '2019-05-17');`

   `INSERT INTO vets (name, age, date_of_graduation)
   VALUES ('Jack Harknes', 38, '2008-06-08'); `

File: data.sql [hunk 2]
```diff
@@ -38,3 +38,11 @@ WHERE Name = 'Charmander' or Name = 'Squirtle' or Name = 'Blossom';
...
+
+INSERT INTO specialization VALUES(1, 1 ),(1, 3 ), (2, 3 ),(2, 4 );
```
Comment (Emmyn5600) [inline data.sql]: - [ ] Let us modify this implementation by selecting id from vets and id from species. Your implementation is very complicated to know what exactly are you trying to insert into database.

Hint:  `INSERT INTO specializations (species_id, vet_id)
 VALUES ((SELECT id FROM vets WHERE name = 'William Tatcher'), (SELECT id FROM species WHERE name = 'Pokemon'));`

File: data.sql [hunk 3]
```diff
@@ -38,3 +38,11 @@ WHERE Name = 'Charmander' or Name = 'Squirtle' or Name = 'Blossom';
...
+
+INSERT INTO visits (animals_id, vet_id, date) VALUES (1, 1, (to_date('05/24/2020','mm/dd/yyyy')) ), (1, 3, (to_date('07/22/2020','mm/dd/yyyy')) ),(2, 4, (to_date('02/02/2021','mm/dd/yyyy')) ),(3, 2, (to_date('01/05/2020','mm/dd/yyyy')) ),(3, 2, (to_date('03/08/2020','mm/dd/yyyy')) ),(3, 2, (to_date('05/14/2020','mm/dd/yyyy')) ),(4, 3, (to_date('05/04/2021','mm/dd/yyyy')) ),(5, 4, (to_date('02/24/2021','mm/dd/yyyy')) ),(7, 2, (to_date('12/21/2019','mm/dd/yyyy')) ),(7, 1, (to_date('08/10/2020','mm/dd/yyyy')) ),(7, 2, (to_date('04/07/2021','mm/dd/yyyy')) ),(7, 3, (to_date('09/29/2019','mm/dd/yyyy')) ),(8, 4, (to_date('10/03/2020','mm/dd/yyyy')) ),(8, 4, (to_date('11/04/2020','mm/dd/yyyy')) ),(9, 2, (to_date('01/24/2019','mm/dd/yyyy')) ),(9, 2, (to_date('05/15/2019','mm/dd/yyyy')) ),(9, 2, (to_date('02/27/2020','mm/dd/yyyy')) ),(9, 2, (to_date('08/03/2020','mm/dd/yyyy')) ),(10, 3, (to_date('05/24/2020','mm/dd/yyyy')) ),(10, 1, (to_date('01/11/2021','mm/dd/yyyy')) );
+
```
Comment (Emmyn5600) [inline data.sql]: - [ ] Let us use this way to insert the data into vets table to keep our code clean and clear as well as easy to read. You need to select from animal and vets tables

File: queries.sql [hunk 4]
```diff
@@ -82,3 +82,55 @@ WHERE Escape_attempts = 0 AND owners.id = 5;
...
+JOIN specialization ON specialization.vets_id = vets.id
+WHERE specialization.specie_id <> animals.species_id;
+
+SELECT species.name,COUNT(visits.animals_id) AS count_species FROM visits
+JOIN vets ON vets.id = visits.vet_id
+JOIN animals ON animals.id = visits.animals_id
+JOIN species ON species.id = animals.species_id
+WHERE vets.id = 2
+GROUP BY species.name
+ORDER BY count_species DESC LIMIT 1;
```
Comment (Emmyn5600) [inline queries.sql:136]: - [x] Make sure the changes you are going to implement , doesn't affect your solution

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUESTED :recycle:

Hi, @pasytchangwa 

Great work on implementing the project requirements, however there are some issues to be solved before you get approved but you are almost there:muscle:
   
 - The repo looks professional.:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

     Changes required :recycle: 
    
    -  Please check the comment under the review too

    Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

File: data.sql (hunk 1 above)
Comment (pasytchangwa) [reply data.sql]: Thanks for the alternative. Done

File: data.sql (hunk 2 above)
Comment (pasytchangwa) [reply data.sql]: Thanks for the alternative. Done

File: data.sql (hunk 3 above)
Comment (pasytchangwa) [reply data.sql]: Thanks for the alternative. Done

File: queries.sql (hunk 4 above)
Comment (pasytchangwa) [reply queries.sql:136]: Sure

**Comment by Davidosky007** [comment]
## Hi @pasytchangwa 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 


job-welldone-gif

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**